core/internal/features_test.go

```diff
@@ -713,6 +713,7 @@ func TestIntegration_SyncJobRuns(t *testing.T) {
 	err := json.Unmarshal([]byte(message), &run)
 	require.NoError(t, err)
 	assert.Equal(t, j.ID, run.JobSpecID)
+	cltest.WaitForJobRunToComplete(t, app.Store, run)
```

Comment:
Ooooh baby. This seems like it might do 'er.

Reply:
Good enough for now, although it seems like a code smell that the application doesn't already wait for job runs to finish upon shutdown.

Reply:
Made a story for this: https://www.pivotaltracker.com/story/show/171577373